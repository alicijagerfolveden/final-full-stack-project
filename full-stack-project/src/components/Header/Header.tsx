import { Box, Typography, Grid, Button } from "@mui/material";
import { useContext } from "react";
import { Link, useNavigate } from "react-router-dom";
import { AuthContext } from "../AuthContext";

export const Header = () => {
  const { auth, setAuth } = useContext(AuthContext);
  const navigate = useNavigate();

  const handleSignOutClick = () => {
    setAuth("");

    sessionStorage.removeItem("accessToken");
  };

  const handleSignInClick = () => {
    navigate("/");
  };

  return (
    <Box
      component="header"
      sx={{ borderBottom: "3px solid gray", marginBottom: "20px" }}
    >
      <Grid container justifyContent="space-around" alignItems="center">
        <Grid item>
          <Box
            component="img"
            sx={{ maxHeight: "200px" }}
            src="https://static.vecteezy.com/system/resources/thumbnails/000/552/753/small/fireworks_005.jpg"
            alt="event logo"
          />
        </Grid>
        <Grid item>
          <Typography variant="h3" component="h1" fontFamily="cursive">
            Make your life happier
          </Typography>
        </Grid>
        <Grid item>
          {auth ? (
            <Button
              variant="contained"
              color="secondary"
              sx={{ width: 200 }}
              onClick={handleSignOutClick}
            >
              Sign out
            </Button>
          ) : (
            <Button
              variant="contained"
              color="secondary"
              sx={{ width: 200 }}
              onClick={handleSignInClick}
            >
              Sign in
            </Button>
          )}
        </Grid>
      </Grid>
      <Box
        component="nav"
        display="flex"
        justifyContent="space-around"
        marginBottom={4}
        marginLeft={10}
        marginRight={10}
        sx={{ borderTop: "3px solid gray", paddingTop: "29px" }}
      >
        <Link
          to="/users"
          aria-label="users link"
          style={{ textDecoration: "none", fontSize: "20px" }}
        >
          Users
        </Link>
        <Link
          to="/register"
          aria-label="register new user link"
          style={{ textDecoration: "none", fontSize: "20px" }}
        >
          Register new user
        </Link>
        <Link
          to="/events"
          aria-label="events link"
          style={{ textDecoration: "none", fontSize: "20px" }}
        >
          Events
        </Link>
      </Box>
    </Box>
  );
};
