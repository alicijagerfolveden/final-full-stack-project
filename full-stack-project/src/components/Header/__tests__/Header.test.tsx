import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { Header } from "../Header";
import renderer from "react-test-renderer";

describe("Header", () => {
  it("should render Header", () => {
    render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );

    expect(screen.getByText("Make your life happier")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );

    expect(tree).toMatchSnapshot();
  });

  it.each(["events link", "users link", "register new user link"])(
    "should render %s",
    (link) => {
      render(
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      );

      expect(screen.getByLabelText(link)).toBeVisible();
    }
  );
});
