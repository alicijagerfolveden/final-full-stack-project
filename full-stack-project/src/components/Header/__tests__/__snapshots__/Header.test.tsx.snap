// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<header
  className="MuiBox-root css-48t7i8"
>
  <div
    className="MuiGrid-root MuiGrid-container css-iq4gde-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <img
        alt="event logo"
        className="MuiBox-root css-1a7twpa"
        src="https://static.vecteezy.com/system/resources/thumbnails/000/552/753/small/fireworks_005.jpg"
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <h1
        className="MuiTypography-root MuiTypography-h3 css-whusbg-MuiTypography-root"
      >
        Make your life happier
      </h1>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-w5oxbz-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Sign in
      </button>
    </div>
  </div>
  <nav
    className="MuiBox-root css-1qpb92y"
  >
    <a
      aria-label="users link"
      href="/users"
      onClick={[Function]}
      style={
        Object {
          "fontSize": "20px",
          "textDecoration": "none",
        }
      }
    >
      Users
    </a>
    <a
      aria-label="register new user link"
      href="/register"
      onClick={[Function]}
      style={
        Object {
          "fontSize": "20px",
          "textDecoration": "none",
        }
      }
    >
      Register new user
    </a>
    <a
      aria-label="events link"
      href="/events"
      onClick={[Function]}
      style={
        Object {
          "fontSize": "20px",
          "textDecoration": "none",
        }
      }
    >
      Events
    </a>
  </nav>
</header>
`;
